core/modules/syncer.js

```diff
@@ -134,6 +135,14 @@ function Syncer(options) {
 			$tw.utils.copyToClipboard($tw.utils.getSystemInfo() + "\n\nLog:\n" + self.logger.getBuffer());
 		});
 	}
+	// Handle loading of skinny tiddlers on demand.
+	$tw.hooks.addHook("th-navigating", (evt) => {
```

Comment:
Ok I'm gonna have to do more digging. Originally when I tried opening a skinny tiddler the `lazyLoad` event didn't fire at all. So that's why I added this navigation hook. But I can look more into the `parseTextReference` you mentioned.

Reply:
Alright I figured out why it wasn't working. Turns out the "checking for the presence of `_is_skinny` OR the lack of `text`" logic needed to be replicated in both `Syncer#syncFromServer` and `Wiki#getTiddlerText`. Previously `Wiki#getTiddlerText` was only checking for the presence of `_is_skinny` without the "lack of `text`" part. That's why the `lazyLoad` event wasn't firing when navigating. I updated `Wiki#getTiddlerText`'s logic to match the logic of `Syncer#syncFromServer` and now the `lazyLoad` event is firing properly.

I also discovered another issue: In `Syncer#handleLazyLoadEvent`, the code is only queuing the title for fetch but did not call `Syncer#syncFromServer` after queuing the title. This means that when navigating to a skinny tiddler, the fat tiddler would only be fetched on the `Syncer` poll interval, which can take a while. So I also updated `Syncer#handleLazyLoadEvent` to immediately call `Syncer#syncFromServer` after queuing the title for fetching.